lib/Service/DiscoveryService.php

```diff
@@ -80,6 +80,16 @@ public function obtainDiscovery(Provider $provider): array {
 			$client = $this->clientService->newClient();
 			$response = $client->get($url);
 			$cachedDiscovery = $response->getBody();
+
+			// Manipulate the response with the custom endpoint url
+			$endpointData = json_encode($this->providerService->getProviderWithSettings($provider->getId()));
+			$endpointData = json_decode($endpointData)->endSessionEndpoint;
+			if ($endpointData) {
+				$discoveryData = json_decode($cachedDiscovery);
+				$discoveryData->end_session_endpoint = $endpointData;
+				$cachedDiscovery = json_encode($discoveryData);
+			}
+
```

Comment:
Replacing/setting the custom end session endpoint in the discovery itself means it is being cached so changing it won't be effective immediately.
Since this value is only used in `LoginController::singleLogoutService`, I think it's better to access it there, directly from the provider object.
Same logic as what you did here: Use the custom value if it's set, fallback on the one from the discovery.

Small detail (not useful here anymore): `json_decode` has a second `associative` boolean parameter. Setting it to true makes the function return an associative array instead of an object. We usually do it this way.

Reply:
Fixed - moved it into the login controller and adjusted the code